libfreshclam/libfreshclam.c

```diff
@@ -123,6 +123,71 @@ const char *fc_strerror(fc_error_t fcerror)
     }
 }
 
+int fc_upsert_logg_file(fc_config *fcConfig)
+{
+    if (fcConfig->logFile == NULL) {
+        return 0;
+    }
+    int ret = 0, field_no          = 1;
+    char *current_path, *file_path = cli_safer_strdup(fcConfig->logFile), *token;
+    FILE *logg_fp = NULL;
+    token         = cli_strtok(file_path, field_no++, PATHSEP);
```

Comment:
I think it would be easier to follow if this line were moved down to just above the `while()` loop. 

On the topic of readability, Some blank lines between unrelated lines of code would be nice on the eyes.